Src/AutoNSubstitute/AutoNSubstitute.csproj

```diff
@@ -64,6 +64,7 @@
   <ItemGroup>
     <Compile Include="AutoConfiguredNSubstituteCustomization.cs" />
     <Compile Include="AutoNSubstituteCustomization.cs" />
+    <Compile Include="AutoSubstituteCustomization.cs" />

```

Comment:
> Their `Builder` inspection properties mayhave to be modified to return an `ISpecimenBuilder` instance of a different type - `SubstituteBuilder` or `SubstituteAttributeRelay` instead of the `NSubstituteBuilder`. The type of property itself would remain an `ISpecimenBuilder`.

> If you consider that change to be *backward compatible*

I do. :smile: If clients use this property and attempt to downcast to the concrete type, they are on their own.

> merging the new `AutoSubstituteCustomization` logic into the two existing `NSubstituteCustomization` classes and eliminating the new customization class before merging this PR;

Sounds promising :+1: 

> replacing the existing `NSubstituteBuilder` implementation with that of the new `SubstituteBuilder` and eliminating the new builder class before merging this PR;

Do they do the same thing? They don't look identical to me, but I'm not so deep in the bowels of this part of the code base as you are. If it's possible to implement one using the other, it sounds like a beneficial refactoring.

Nevertheless, unless it's *necessary* to do as part of this PR, I would prefer that it's not made a part of this PR. [Keep pull requests as small and focused as possible](http://blog.ploeh.dk/2015/01/15/10-tips-for-better-pull-requests).

> renaming the new SubstituteXxx classes to follow the existing NSubstituteXxx naming convention.

*That* would be a breaking change, so unfortunately, we can't do that.

What we *can* do is to copy the old class to a new class with the desired name, and then deprecate the old class, but that's not going to make the number of classes smaller. Again, if we do that, we should do that in a separate pull request.